src/Core/Tool/Uploader.php

```diff
@@ -92,6 +92,17 @@ public function upload(UploadData $file, $filename = null)
 			]);
 	}
 
+	/**
+	 * Clean up file if upload failed
+	 * @param  $file  filepath
+	 */
+	public function delete($file)
```

Comment:
Its plausible that now I've added this method, I should really rename this to `FileActions` or something. But I don't really care that much :)